### PR TITLE
Refactor to use common function to attach buttons

### PR DESCRIPTION
**Pull request by jonrandy**
**1. Objective**

Refactoring so all 'attaching' of buttons is done with the same code, instead of two near identical pieces of code for it.

**2. Description of change**

Added a generic `_attachButton` method to do all the work. Modified other places accordingly to use it.

**3. Quality assurance**

Check everything still works as expected, and tests still pass

**4. Operations impact**

None

### CODE REVIEW

File: src/OmiseCard.js [hunk 1]
```diff
@@ -453,6 +428,43 @@ export default function OmiseCardFactory(settings, initWhenStart = true) {
...
+    let buttonText = _app.defaultConfig.buttonLabel
+
+    if (cfg.buttonLabel && buttonText !== cfg.buttonLabel) {
+      buttonText = cfg.buttonLabel
+    } else if (button.innerHTML) {
+      buttonText = button.innerHTML
+    }
+
+    button.innerHTML = buttonText
+    button.className += (button.className && ' ') + 'omise-checkout-button'
```
Comment (ratchagarn) [inline src/OmiseCard.js:445]:
```suggestion
    button.className += `${button.className} omise-checkout-button`
```

File: src/OmiseCard.js [hunk 2]
```diff
@@ -258,74 +257,48 @@ export default function OmiseCardFactory(settings, initWhenStart = true) {
...
-    } else {
-      console.warn('Missing Omise script tag');
-    }
-
-    // bind button event.
-    checkoutButton.addEventListener(
-      'click',
-      event => {
-        event.preventDefault()
+    _attachButton('#' + checkoutButton.id, config)
```
[the remark this answers is not in the document]
Comment (ratchagarn) [reply src/OmiseCard.js:293]:
```suggestion
    _attachButton(`#${checkoutButton.id}`, config)
```

File: src/OmiseCard.js [hunk 3]
```diff
@@ -283,40 +283,15 @@ export default function OmiseCardFactory(settings, initWhenStart = true) {
...
-    )
-
-    // inject button next script tag.
+    checkoutButton.id = '__generated_omise_button'
     omiseScriptTag.parentNode.insertBefore(
       checkoutButton,
       omiseScriptTag.nextSibling
     )
 
+    _attachButton('#' + checkoutButton.id, config)
```
Comment (ratchagarn) [inline src/OmiseCard.js:293]:
```suggestion
    _attachButton(`#${checkoutButton.id}`, config)
```

File: src/OmiseCard.js (hunk 1 above)
Comment (ratchagarn) [reply src/OmiseCard.js:445]:
```suggestion
    button.className += `${(button.className && ' ')omise-checkout-button`
```

Comment (jonrandy) [reply src/OmiseCard.js:445]: They're the same length and achieve the same thing. The original is also easier to read - the second looks cluttered

File: src/OmiseCard.js (hunk 3 above)
Comment (jonrandy) [reply src/OmiseCard.js:293]: Same as above

File: src/OmiseCard.js (hunk 1 above)
Comment (ratchagarn) [reply src/OmiseCard.js:445]: Hmm, You have a point

**Pull request merged**